### PR TITLE
[Bindless] Struct for unique addressing modes per dimension

### PR DESCRIPTION
**Pull request by Seanst98**
This MR adds a struct allowing for unique addressing modes per dimension

### CODE REVIEW

**Review by veselypeta — APPROVED** [review]

**Review by kbenzie — APPROVED** [review]

File: scripts/core/exp-bindless-images.yml [hunk 1]
```diff
@@ -176,6 +179,19 @@ members:
...
+desc: "Describes unique sampler addressing mode per dimension"
+details:
+    - Specify these properties in $xSamplerCreate via $x_sampler_desc_t as part
+      of a `pNext` chain.
+class: $xBindlessImages
+name: $x_exp_sampler_addr_modes_t
+base: $x_base_properties_t
+members:
+    - type: $x_sampler_addressing_mode_t
+      name: addrModes[3]
```
Comment (veselypeta) [inline scripts/core/exp-bindless-images.yml:191]: Unfortunatelly this won't work with the current generator. I think the simplest solution would be to just have three named fields for the addressing mode values.

File: scripts/core/exp-bindless-images.yml [hunk 2]
```diff
@@ -176,6 +179,19 @@ members:
...
+      desc: "[in] Specify the address mode of the sampler per dimension"
```
Comment (veselypeta) [inline scripts/core/exp-bindless-images.yml:192]:
```suggestion
    - type: $x_sampler_addressing_mode_t
      name: addModeX
      desc: "[in] Specify the addressing mode of the x-dimension."
    - type: $x_sampler_addressing_mode_t
      name: addModeY
      desc: "[in] Specify the addressing mode of the y-dimension."
    - type: $x_sampler_addressing_mode_t
      name: addModeX
      desc: "[in] Specify the addressing mode of the x-dimension."
```

Comment (Seanst98) [reply scripts/core/exp-bindless-images.yml]: I've made the change.

File: scripts/core/exp-bindless-images.yml (hunk 1 above)
Comment (veselypeta) [reply scripts/core/exp-bindless-images.yml]: I've created https://github.com/oneapi-src/unified-runtime/issues/869 to add this support.

Comment (Seanst98) [reply scripts/core/exp-bindless-images.yml]: Great, thanks! We'll keep an eye on this but will use the separate variables as a temporary solution.

**Pull request merged**